cpp/src/arrow/dataset/file_csv.h

```diff
@@ -61,6 +61,10 @@ class ARROW_DS_EXPORT CsvFileFormat : public FileFormat {
       const std::shared_ptr<ScanOptions>& scan_options,
       const std::shared_ptr<FileFragment>& file) const override;
 
+  Future<util::optional<int64_t>> CountRows(
+      const std::shared_ptr<FileFragment>& file, compute::Expression predicate,
+      std::shared_ptr<ScanOptions> options) override;
```

Comment:
Is there a reason that some methods take `const std::shared_ptr<ScanOptions>&` and this one `std::shared_ptr<ScanOptions>`?

Reply:
I think I had patterned everything on Fragment::Scan (which takes it by value) but all the new methods take it by const reference so I'll adjust that.